src/librbd/crypto/CryptoObjectDispatch.cc

```diff
@@ -587,8 +587,8 @@ bool CryptoObjectDispatch<I>::discard(
 
   *dispatch_result = io::DISPATCH_RESULT_COMPLETE;
   auto req = io::ObjectDispatchSpec::create_write_same(
-          object_len, {{0, buffer_size}}, std::move(bl), io_context,
           m_image_ctx, io::OBJECT_DISPATCH_LAYER_CACHE, object_no, object_off,
+          object_len, {{0, object_len}}, std::move(bl), io_context,
```

Comment:
I don't understand.
This change is a bug in the implementation of Discard which has nothing to do with the cache deadlocks.

Reply:
This compile error that you are fixing here was introduced in the previous commit -- so instead of having one commit that creates a compile error and the next that fixes, just squash the two together to avoid introducing a compile error.

Reply:
The change I'm seeing here is ``{{0, object_len}}`` instead of ``{{0, buffer_size}}``.
This is not a fix for a compilation error, but rather for a runtime bug in Discard.
Are you referring to a different change?

Reply:
Yes -- you moved the line in the previous commit (see how you deleted line 589 and inserted at line 591.

Reply:
Got you! Thanks!!